Assets/Scripts/Behaviours/World/DynamicObject.cs

```diff
@@ -0,0 +1,171 @@
+            if (properties.HasValue)
+            {
+                m_breakableObject.DynamicObjectProperties = properties.Value;
+            }
+            else // if non gta model get loaded
+            {
+                m_breakableObject.DynamicObjectProperties = new DynamicObjectProperties
+                {
+                    mass = 100,
+                    breakImpulse = 10000,
+                    canBeCrashedByPed = false,
+                    canBeCrashedByVehicle = false,
+                    canBeShooted = false,
+                    health = 1000,
+                };
+            }
+            m_damageable.MaxHealth = m_breakableObject.DynamicObjectProperties.health;
+            m_damageable.Health = m_damageable.MaxHealth;
+        }
+
+        void OnDamage()
+        {
+            if (m_breakableObject.IsBroken)
+                return;
+
+            DamageInfo damageInfo = m_damageable.LastDamageInfo;
+
+            if (m_breakableObject.DynamicObjectProperties.canBeShooted)
+                m_damageable.Health -= damageInfo.amount;
+
+            if(m_damageable.Health > 0.0f)
+            {
+                m_breakableObject.PlayWeaponDamageEffect();
+                Rigidbody rb = gameObject.GetComponent<Rigidbody>();
+                if (rb.IsSleeping())
+                    rb.WakeUp();
+                rb.AddForceAtPosition(-damageInfo.hitNormal * 2, damageInfo.hitPoint - transform.position); // @TODO fix calculations
+                //rb.AddForceAtPosition(damageInfo.hitPoint - transform.position , - damageInfo.hitNormal * 2);
+            }
+            else
+            {
+                m_breakableObject.Break();
+            }
+        }
+
+        protected override void OnLoad()
+        {
+            if (!m_loaded)
+            {
+                m_breakableObject.RespawnPosition = transform.position;
+                m_breakableObject.RespawnRotation = transform.rotation.eulerAngles;
+
+                m_breakableObject.BreakEffect = ParticleSystemManager.Instance.GetByNane("Debris");
+                m_breakableObject.BreakEffect.transform.SetParent(transform, false);
+                m_breakableObject.BreakEffect.transform.localPosition = Vector3.zero;
+                m_breakableObject.BreakEffect.transform.localRotation = Quaternion.identity;
+                m_breakableObject.BreakEffect.name += " breakEffect";
+                m_breakableObject.WeaponDamageEffect = ParticleSystemManager.Instance.GetByNane("Debris");
+                m_breakableObject.WeaponDamageEffect.transform.SetParent(transform, false);
+                m_breakableObject.WeaponDamageEffect.transform.localPosition = Vector3.zero;
+                m_breakableObject.WeaponDamageEffect.transform.localRotation = Quaternion.identity;
+                m_breakableObject.WeaponDamageEffect.name += " weaponDamageEffect";
+                InvokeRepeating("CheckForRespawn", 1f, 1f);
```

Comment:
This costs performance. There are a lot of dynamic objects on the map. But anyway, this should be removed because it is not the correct way to spawn dynamic objects.